### PR TITLE
WidgetBuilder: Live Config Editor

### PR DESCRIPTION
**Pull request by tokdaniel**
As we have to do a two way mapping on DisplaySettings <-> ThemeOptions, this has some challenges, thus the theme is not displayed yet in the config editor. Otherwise, it's 2way edit, so if you change something on the UI it will be reflected in the config, and vice versa.

### CODE REVIEW

File: apps/widget-builder/src/components/config-editor/ConfigEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+type ConfigEditorProps = {
+  value: WidgetProps;
+  setValue: UseFormSetValue<WidgetProps>;
+};
+
+const schema = z.object({
+  productDetails: productDetailsSchema,
+  paymentDetails: z.object({
+    defaultReceiverAddress: z.string().startsWith("0x").length(42),
```
[the remark this answers is not in the document]
Comment (tokdaniel) [reply apps/widget-builder/src/components/config-editor/ConfigEditor.tsx:20]: We should probably add a bool flag though, to use something as a default payment option.

File: apps/widget-builder/src/components/config-editor/ConfigEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    ...paymentDetailsSchema.shape,
+  }),
+  layout: z.enum(["dialog", "drawer", "full-screen", "page"]),
+  displaySettings: z.object({
```
[the remark this answers is not in the document]
Comment (tokdaniel) [reply apps/widget-builder/src/components/config-editor/ConfigEditor.tsx:24]: The thing is we probably want to validate the scheme of the theme, so I wonder if it's available somewhere

File: apps/widget-builder/src/components/config-editor/ConfigEditor.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (kasparkallas) [reply apps/widget-builder/src/components/config-editor/ConfigEditor.tsx:20]: I don't see a reason to make the schema more complex right now with the normalization/fallback logic.

The "default address" convenience can be kept only in the Builder.

File: apps/widget-builder/src/components/config-editor/ConfigEditor.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kasparkallas) [reply apps/widget-builder/src/components/config-editor/ConfigEditor.tsx:24]: Technically the TypeScript intellisense could be enabled based on the types. It's a web worker based logic from my understanding. Might be a bit of a rabbithole.

There should be an example here:
https://github.com/microsoft/monaco-editor/tree/main/samples

It's a nice-to-have thing.

I'd rather just disable the intellisense for now than create the intermediary schemas.

File: apps/widget-builder/src/components/config-editor/ConfigEditor.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (tokdaniel) [reply apps/widget-builder/src/components/config-editor/ConfigEditor.tsx]: I'd say we should follow the widget type as closely as we can, so removing the defaultReceiver would be the best until we support it in the widget

**Review by elvijsTDL — APPROVED** [review]
Has room for improvments, but overall LGTM 👍 

For a perfect editor id like it to also validate the super token address and highlight that there is an issue
But i’m not finding anything that crashes the page yet
Also in a perfect world if I change the only payment option, it should pick it up in the widget too instead of getting stuck on the last thing I edited
And most of the payment validation errors are not showing up as highlighted issues in the json editor but a quick error pops up and disappears
If you change the period away from month the wrap amount should change up to months worth of tokens instead of the ether amount

**Pull request merged**